Improve Type Assertion Handling in `Compiler.Compile`

Hi Permify team,

I recently discovered your project, and it looks very promising! 
I'm interested in contributing and have started exploring the code.

While reviewing the `Compile` function in `pkg/dsl/compiler/compiler.go`, I noticed an opportunity for improvement regarding type assertions.

In this section: [link to code at line 49](https://github.com/Permify/permify/blob/a80acc852a20959fe985519d6510912c91e0b9d0/pkg/dsl/compiler/compiler.go#L49)

Currently, the type assertion result is used multiple times inside the `switch` cases. To make the code cleaner and more efficient, we could assign the result of the type assertion to a variable once and then use that variable inside the `switch`. This would eliminate the need for multiple type assertions.

The `staticcheck` tool also reports this issue:

```
pkg/dsl/compiler/compiler.go:49:10: assigning the result of this type assertion to a variable (switch statement := statement.(type)) could eliminate type assertions in switch cases (S1034)
pkg/dsl/compiler/compiler.go:52:27: could eliminate this type assertion
pkg/dsl/compiler/compiler.go:68:25: could eliminate this type assertion
```

_**If you agree with this suggestion, I'd be happy to take care of the fix. Please feel free to assign the issue to me!**_

Thanks!

Note: even if we keep the assertions in the `switch` cases, we don't need the `ok` check (`v, ok := s.(type)`). We can simplify this to just `v := s.(type)` since the assertion is already performed. So for example, this check will never evaluate to `true`, since `ok` is always `true`: [link to code at line 53](https://github.com/Permify/permify/blob/a80acc852a20959fe985519d6510912c91e0b9d0/pkg/dsl/compiler/compiler.go#L53). But anyway, assigning the result of the type assertion to a variable will eliminate the need for both, we will no longer need to do type assertions again in `switch cases`.

--- a/pkg/dsl/compiler/compiler.go
+++ b/pkg/dsl/compiler/compiler.go
@@ -46,17 +46,10 @@ func (t *Compiler) Compile() ([]*base.EntityDefinition, []*base.RuleDefinition, 
 
 	// Loop through each statement in the schema.
 	for _, statement := range t.schema.Statements {
-		switch statement.(type) {
+		switch v := statement.(type) {
 		case *ast.EntityStatement:
-			// Check if the statement is an EntityStatement.
-			entityStatement, ok := statement.(*ast.EntityStatement)
-			if !ok {
-				// If the statement is not an EntityStatement, return a compile error.
-				return nil, nil, compileError(entityStatement.Entity.PositionInfo, base.ErrorCode_ERROR_CODE_SCHEMA_COMPILE.String())
-			}
-
 			// Compile the EntityStatement into an EntityDefinition.
-			entityDef, err := t.compileEntity(entityStatement)
+			entityDef, err := t.compileEntity(v)
 			if err != nil {
 				return nil, nil, err
 			}
@@ -64,15 +57,8 @@ func (t *Compiler) Compile() ([]*base.EntityDefinition, []*base.RuleDefinition, 
 			// Append the EntityDefinition to the slice of entity definitions.
 			entities = append(entities, entityDef)
 		case *ast.RuleStatement:
-			// Check if the statement is a RuleStatement.
-			ruleStatement, ok := statement.(*ast.RuleStatement)
-			if !ok {
-				// If the statement is not a RuleStatement, return a compile error.
-				return nil, nil, compileError(ruleStatement.Rule.PositionInfo, base.ErrorCode_ERROR_CODE_SCHEMA_COMPILE.String())
-			}
-
 			// Compile the RuleStatement into a RuleDefinition.
-			ruleDef, err := t.compileRule(ruleStatement)
+			ruleDef, err := t.compileRule(v)
 			if err != nil {
 				return nil, nil, err
 			}

